Documentation/requirements.md

```diff
@@ -0,0 +1,29 @@
+# Requirements
+
+## Ports
+
+The information below describes a minimum set of port allocations used by Kubernetes components.
+
+### Master node(s) ingress
+
+| Protocol | Port Range | Source                                    | Purpose                |
+-----------|------------|-------------------------------------------|------------------------|
+| TCP      | 443        | Worker Nodes, API Requests, and End-Users | Kubernetes API server. |
+| UDP      | 8472       | Master & Worker Nodes                     | flannel overlay network - *vxlan backend* |
+
+### etcd node(s) ingress
+
+| Protocol | Port Range | Source                | Purpose                                          |
+-----------|------------|-----------------------|--------------------------------------------------|
+| TCP      | 2379-2380  | Master & Worker Nodes | etcd server client API                           |
+
+### Worker node(s) ingress
+
+| Protocol | Port Range  | Source                         | Purpose                                                                |
+-----------|-------------|--------------------------------|------------------------------------------------------------------------|
+| UDP/TCP  | 53          | Master & Worker Nodes          | Internal DNS server |
+| UDP      | 8472        | Master & Worker Nodes          | flannel overlay network - *vxlan backend* |
+| TCP      | 10249       | Health check                   | Kube proxy health check                                  |
+| TCP      | 10250       | Master Nodes                   | Worker node Kubelet API for exec and logs.                                  |
+| TCP      | 10255       | Heapster                       | Worker node read-only Kubelet API.                                  |
```

Comment:
heapster could be on workers or master (right now at least) - might make more clear to just note those as the sources (rather than the app name).

Also, it has been a while since I've looked but I thought heapster itself needed access to a cadvisor port (but maybe not anymore)?